src/components/enterprise-user-subsidy/data/utils.js

```diff
@@ -15,3 +18,49 @@ export const transformRedeemablePoliciesData = (policies) => policies?.map(polic
     learner_content_assignments: assignmentsWithSubsidyExpiration,
   };
 });
+
+/**
+ * Determine whether learner has only content assignments available to them, based on the presence of:
+ * - content assignments for display (allocated or canceled)
+ * - no auto-applied budgets
+ * - no current enterprise offers
+ * - no active license or license requests
+ * - no assigned codes or code requests
+ *
+ * @param {Object} params - The parameters object.
+ * @param {Object} params.subscriptionPlan - The subscription plan of the learner.
+ * @param {Object} params.subscriptionLicense - The subscription license of the learner.
+ * @param {Array} params.licenseRequests - The license requests of the learner.
+ * @param {number} params.couponCodesCount - The count of assigned coupon codes of the learner.
+ * @param {Array} params.couponCodeRequests - The coupon code requests of the learner.
+ * @param {Object} params.redeemableLearnerCreditPolicies - The redeemable learner credit policies.
+ * @param {boolean} params.hasCurrentEnterpriseOffers - Whether the learner has current enterprise offers.
+ * @returns {boolean} - Returns true if the learner has only content assignments available to them, false otherwise.
+ */
+export const determineLearnerHasContentAssignmentsOnly = ({
+  subscriptionPlan,
+  subscriptionLicense,
+  licenseRequests,
+  couponCodesCount,
+  couponCodeRequests,
+  redeemableLearnerCreditPolicies,
+  hasCurrentEnterpriseOffers,
+}) => {
+  const hasActiveLicense = !!(subscriptionPlan?.isActive && subscriptionLicense?.status === LICENSE_STATUS.ACTIVATED);
+  const hasActiveLicenseOrLicenseRequest = hasActiveLicense || licenseRequests.length > 0;
+  const hasAssignedCodesOrCodeRequests = couponCodesCount > 0 || couponCodeRequests.length > 0;
```

Comment:
[curious] compared to the original `isDisableCourseSearch` the conditional was originally checking whether there was an `available` coupon code. Just wondering if we should do the same there 🤔

Reply:
Great question! We discussed synchronously in Slack, but posting here for posterity/future us:

In `useCouponCodes`, we make 2 API calls to ecommerce:
- `/api/v2/enterprise/offer_assignment_summary/`
- `/api/v2/enterprise/coupons/:enterpriseUUID/overview/`

The former API endpoint is passed a query parameter `?is_active=True` which forces the API to only return available coupon codes in its response. In practice, that extra check on `available` previously was always going to be truthy. The latter API endpoint is aware of expired/inactive coupon/code batches; it's what is used to power the "your code(s) expired" messaging on the course page, when relevant.